File: sunflower-core/src/main/java/org/tdf/sunflower/net/PluginWrapper.java
package org.tdf.sunflower.net;

import org.tdf.sunflower.facade.PeerServerListener;
import org.tdf.sunflower.proto.Code;

// wrap listener as plugin
public class PluginWrapper implements Plugin {
    private final PeerServerListener listener;

    PluginWrapper(PeerServerListener listener) {
        this.listener = listener;
    }

    @Override
    public void onMessage(ContextImpl context, PeerServerImpl server) {
        if (context.message.getCode().equals(Code.ANOTHER)) {
            listener.onMessage(context, server);
        }
    }

    @Override
    public void onStart(PeerServerImpl server) {
        listener.onStart(server);
    }

    @Override
    public void onNewPeer(PeerImpl peer, PeerServerImpl server) {
        listener.onNewPeer(peer, server);
    }

    @Override
    public void onDisconnect(PeerImpl peer, PeerServerImpl server) {
        listener.onDisconnect(peer, server);
    }

    @Override
    public void onStop(PeerServerImpl server) {

    }
}

File: sunflower-core/src/main/java/org/tdf/sunflower/net/PluginWrapper.kt
package org.tdf.sunflower.net

import org.tdf.sunflower.facade.PeerServerListener
import org.tdf.sunflower.proto.Code

// wrap listener as plugin
class PluginWrapper internal constructor(private val listener: PeerServerListener) : Plugin {
    override fun onMessage(context: ContextImpl, server: PeerServerImpl) {
        if (context.msg.code == Code.ANOTHER) {
            listener.onMessage(context, server)
        }
    }

    override fun onStart(server: PeerServerImpl) {
        listener.onStart(server)
    }

    override fun onNewPeer(peer: PeerImpl, server: PeerServerImpl) {
        listener.onNewPeer(peer, server)
    }

    override fun onDisconnect(peer: PeerImpl, server: PeerServerImpl) {
        listener.onDisconnect(peer, server)
    }

    override fun onStop(server: PeerServerImpl) {}
}